secondarylib-feedback/src/main/scala/com/socrata/datacoordinator/secondary/feedback/FeedbackContext.scala

```diff
@@ -0,0 +1,346 @@
+      case (targetCol, strat) if targetColumns.contains(targetCol) & computationHandler.matchesStrategyType(strat.strategyType) =>
+        computationHandler.setupColumn(perDatasetData, strat, targetCol)
+    }
+    val toCompute = rows.iterator.zipWithIndex.flatMap { case (row, index) =>
+      val rcis =
+        perColumnData.flatMap { columnData =>
+          // don't compute if there has been to change to the source columns
+          if (noChange(row, columnData.strategy.sourceColumnIds))
+            None
+          else
+            Some(computationHandler.setupCell(columnData, row.data))
+        }
+      if (rcis.isEmpty) Iterator.empty
+      else Iterator.single(index -> rcis)
+    }.toMap
+
+    val results = computationHandler.compute(toCompute)
+
+    results.flatMap { case (rowIdx, updates) =>
+      val row = rows(rowIdx)
+      val filtered = updates.filter { case (colId: UserColumnId, value) =>
+        extractCV(row.data, colId) != Some(value) // don't update to current value
+      }
+      val rowId = extractCV(row.data, cookie.primaryKey) match {
+        case None => throw new Exception(s"Cannot find value for primary key ${cookie.primaryKey} in row: $row!")
+        case Some(other) => other
+      }
+      if (filtered.nonEmpty) {
+        val newRow = Map(
+          cookie.primaryKey -> rowId
+        ) ++ filtered.map { case (id, value) => (id, value)}.toMap
+        Some(rowIdx -> newRow)
+      } else {
+        None
+      }
+    }
+  }
+
+  private def noChange(row: Row[CV], columns: Seq[UserColumnId]): Boolean = row.oldData match {
+    case Some(old) =>
+      columns.forall { id =>
+        val internal = cookie.columnIdMap(id)
+        row.data(internal) == old(internal) // safe because updates contain _all_ row values (and this must be one)
+      }
+    case None => false
+  }
+
+  private def extractCV(row: secondary.Row[CV], userId: UserColumnId): Option[CV] = {
+    val internal = cookie.columnIdMap(userId)
+    row.get(internal)
+  }
+
+  // commands for mutation scripts
+  private val commands: Seq[JValue] =
+    j"""[ { "c" : "normal", "user" : $user }
+        , { "c" : "row data", "update_only" : true, "nonfatal_row_errors" : [ "insert_in_update_only" ] }
+        ]""".toSeq
+
+  private def writeMutationScript(rowUpdates: Iterator[JValue]): Option[JArray] = {
+    if (!rowUpdates.hasNext) None
+    else Some(JArray(commands ++ rowUpdates))
+  }
+
+  // this may throw a ReplayLaterSecondaryException or a BrokenDatasetSecondaryException
+  def feedback(rows: Iterator[Row[CV]],
+               targetColumns: Set[UserColumnId] = cookie.strategyMap.keySet,
+               resync: Boolean = false): FeedbackCookie = {
+    val width = cookie.columnIdMap.size
+    val size = batchSize(width)
+
+    log.info("Feeding back rows with batch_size = {} for target computed columns: {}", size, targetColumns)
+
+    def b(retriesLeft: Int, message: String): Int = {
```

Comment:
I can see a name like that if it's used many times, but I don't ese that this is used at all...